lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
I feel like passing extra information to the file icon service needs to be done in a more generic manner, without tying such state to a specific package: I believe we should design services to be general enough to work with other packages that may not necessarily fit into the `tab` and `tree-view` categories. Passing `this` looks somewhat suspicious too, because I am still unclear on the interface that `target` should comply to in order to be passed to `iconClassForPath`.

After taking a look at https://github.com/DanBrooker/file-icons/blob/29218e42fbdbd9d877ec9eb5c3a76848f282e016/lib/service/icon-service.coffee#L146, I feel like we could circumvent this in two ways:

1. For symlinks, query the filesystem in the package itself. That is, call https://nodejs.org/api/fs.html#fs_class_fs_stats and check whether the file is a symlink or not.
2. For the `Show In Tabs` setting, put a configuration setting into each package.

Fundamentally I think there isn't any knowledge (that isn't also available elsewhere) that each package needs to supply to file icon services, and passing state around looks more error-prone than letting the service retrieve pieces of information for itself. Later, when we'll have a better abstraction for file system nodes, we might expose a method called `iconClassForFileSystemNode(node)`, and `node` might have a bunch of facility utilities (such as `isSymlink`) that will allow us to remove some duplication. 

For now, however, it seems like the above proposal should provide a high degree of freedom in the implementation of the service, which makes this design choice more appealing to me.